storage/innobase/srv/srv0srv.cc

```diff
@@ -1421,6 +1421,32 @@ srv_export_innodb_status(void)
 		btr_scrub_total_stat(&scrub_stat);
 	}
 
+#ifdef BTR_CUR_HASH_ADAPT
+	export_vars.innodb_mem_adaptive_hash = 0;
+	ut_ad(btr_search_sys->hash_tables);
+	for (ulong i = 0; i < btr_ahi_parts; i++) {
+		hash_table_t*	ht = btr_search_sys->hash_tables[i];
+
+		ut_ad(ht);
+		ut_ad(ht->heap);
+		/* Multiple mutexes/heaps are currently never used for adaptive
+		hash index tables. */
+		ut_ad(!ht->n_sync_obj);
+		ut_ad(!ht->heaps);
+
+		export_vars.innodb_mem_adaptive_hash += 
+			(ulong) mem_heap_get_size(ht->heap);
+		export_vars.innodb_mem_adaptive_hash += 
+			(ulong) ht->n_cells * sizeof(hash_cell_t);
+	}
+#endif
+
+	mutex_enter(&dict_sys.mutex);
+	export_vars.innodb_mem_dictionary = (dict_sys.table_hash->n_cells + 
+		dict_sys.table_id_hash->n_cells + dict_sys.get_temp_id_hash()->n_cells) 
+		* sizeof(hash_cell_t) + dict_sys_get_size();
+	mutex_exit(&dict_sys.mutex);
```

Comment:
Instead of accessing the internals of `dict_sys_t` here, I would suggest defining and invoking an inline member function that returns the memory usage. The function `get_temp_id_hash()` that you added is unnecessarily exposing an internal data structure.

Reply:
I split it up into 
```
	export_vars.innodb_mem_dictionary = dict_sys.get_cells_size() + 
		dict_sys_get_size();
```